packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -284,24 +295,20 @@ class FocusAttachment {
 ///     _nodeAttachment.reparent();
 ///     return GestureDetector(
 ///       onTap: () {
-///         if (_node.hasFocus) {
-///           setState(() {
+///         if (_focused) {
 ///             _node.unfocus();
```

Comment:
Just curious, why was this originally in a `setState()`?

Reply:
Because it was changing the state, so in order for the component to be rebuilt (since it wasn't listening to focus changes), it needed the `setState`. Now that it's listening for changes, it doesn't need it.

However, most likely the tree would have been rebuilt even without the `setState`, because there's probably a `Focus` widget in the tree above this that would cause the tree to be rebuilt when its focus changed anyhow.